concourse/setup.py

```diff
@@ -100,19 +103,27 @@ def create_image_pull_secret(
 
 @ensure_annotations
 def create_tls_secret(
-    tls_config: TlsConfig,
-    tls_secret_name: str,
+    config_factory: ConfigFactory,
```

Comment:
please revert to passing tls_config + tls_secret_name; extend e.g. with initialised ctx;